v7/navstories/navstories.py

```diff
@@ -26,24 +26,146 @@
+        """
+        ret = []
+        for topent in entries:
+            # Determine toplevel menu name
+            # - If not None, use the name
+            # - If None, use title of page if page with navpath length=1 exist, else navpath[0]
+            title = topent[0]
+            if not title:
+                # Default is 1th level of navpath
+                title = topent[1][0][0][0]
+                # Search for toplevel page (navpath length = 1
+                for i in range(len(topent[1])):
+                    if len(topent[1][i][0]) == 1:
+                        title = topent[1][i][2] # Page Title
+            if len(topent[1]) == 1 and len(topent[1][0][0]) == 1:
+                # Only one menu item and it is not a subpage, let the item go direct to top level menu
+                #ret += tuble(topent[1][0][1], title)
+                pass
+            else:
+                sub = []
+                # Find min/max depth in actual submenu
+                min_depth = 100 # High initial value, set to min(len(navpath)) in this submen
+                max_depth = 0 # Low initial value, set to max(len(navpath)) in this submen
+                for p in topent[1]:
+                    min_depth = min(min_depth, len(p[0]))
+                    max_depth = max(max_depth, len(p[0]))
+                # Map pages to submenu
+                for p in sorted(topent[1], key=lambda x: x[1]): # Sort by second element (permalink) in page list
+                    prefix = self.navstories_submenu_indention * (len(p[0]) - min_depth)
+                    sub.append(tuple([p[1], prefix + p[2]]))
+                ret.append(tuple([tuple(sub), title]))
+        return tuple(ret)
+
+
     def set_site(self, site):
+        """
+        Map navstories config to nav_config[*] as TranslatableSettings
+        """
+
+        # Read NAVSTORIES_SUBMENU_INDENTION and store in self.navstories_submenu_indention
+        if 'NAVSTORIES_SUBMENU_INDENTION' in site.config:
+            self.navstories_submenu_indention = site.config['NAVSTORIES_SUBMENU_INDENTION']
+
+        nav_config = {}
+        for i in self.conf_vars:
+            # Read config variables in a try...except in case a varible is missing
+            try:
+                nav_config[i] = utils.TranslatableSetting(i, site.config[i], site.config['TRANSLATIONS'])
+            except KeyError:
+                # Initialize to "empty" in case config variable i is missng
+                nav_config[i] = utils.TranslatableSetting(i, {}, site.config['TRANSLATIONS'])
+
         site.scan_posts()
         # NAVIGATION_LINKS is a TranslatableSetting, values is an actual dict
         for lang in site.config['NAVIGATION_LINKS'].values:
-            new = []
+            # navstories config for lang
+            nav_conf_lang = {}
+            for i in self.conf_vars:
+                 nav_conf_lang[i] = nav_config[i].values[lang]
```

Comment:
`for` → can’t be changed, not iterable
other instances where there is no iteration: go ahead